modules/core/src/main/java/org/apache/ignite/internal/binary/BinarySerializedFieldComparator.java

```diff
@@ -247,55 +246,16 @@ public static boolean equals(BinarySerializedFieldComparator c1, BinarySerialize
                 Object val1 = c1.currentField();
                 Object val2 = c2.currentField();
 
-                return isArray(val1) ? compareArrays(val1, val2) : F.eq(val1, val2);
+                return isArray(val1) ? F.arrayEq(val1, val2) : F.eq(val1, val2);
```

Comment:
Should we remane `BinarySerializedFieldComparator#isArray()` to something like `isArrayOrBinaryArray()` or fix its description? It looks a bit wierd compared to F.isArray().